docs/LoongArch-ELF-ABI-EN.adoc

```diff
@@ -657,3 +657,191 @@ with check 32-bit unsigned overflow
+When passed in registers or on the stack, the unsigned integer scalars narrower than GRLEN bits are zero-extended to GRLEN bits, and the signed integer scalars are sign-extended.
+
+. GRLEN < WOA ≤ 2 × GRLEN
+
+.. The argument is passed in a pair of GAR, with the low-order GRLEN bits in the lower-numbered register and the high-order GRLEN bits in the higher-numbered register.
+If exactly one register is available, the low-order GRLEN bits are passed in the register and the high-order GRLEN bits are passed on the stack.
+If no GAR is available, it’s passed on the stack.
+
+. WOA > 2 × GRLEN
+
+.. Scalars are passed by reference and are replaced in the argument list with the address.
+
+==== Structure
+
+Empty structures are ignored by C compilers which support them as a non-standard extension(same as union arguments and return values).
+Bits unused due to padding, and bits past the end of a structure whose size in bits is not divisible by GRLEN, are undefined.
+And the layout of the structure on the stack is consistent with that in memory.
+
+. 0 < WOA ≤ GRLEN
+
+.. The structure has only fixed-point members.
+If there is an available GAR, the structure is passed through the GAR by value passing; If no GAR is available, it’s passed on the stack.
+
+.. The structure has only floating-point members:
+
+... One floating-point member.
+The argument is passed in a FAR; If no FAR is available, the value is passed in a GAR; if no GAR is available, the value is passed on the stack.
+
+... Two floating-point members.
+The argument is passed in a pair of available FAR, with the low-order float member bits in the lower-numbered FAR and the high-order float member bits in the higher-numbered FAR.
+If the number of available FAR is less than 2, it’s passed in a GAR, and passed on the stack if no GAR is available.
+
+.. The structure has both fixed-point and floating-point members, i.e. the structure has one float member and
+
+... Multiple fixed-point members.
+If there are available GAR, the structure is passed in a GAR, and passed on the stack if no GAR is available.
+
+... Only one fixed-point member.
+If one FAR and one GAR are available, the floating-point member of the structure is passed in the FAR, and the integer member of the structure is passed in the GAR; If no floating-point register but one GAR is available, it’s passed in GAR; If no GAR is available, it’s passed on the stack.
+
+. GRLEN < WOA ≤ 2 × GRLEN
+
+.. Only fixed-point members.
+
+... The argument is passed in a pair of available GAR, with the low-order bits in the lower-numbered GAR and the high-order bits in the higher-numbered GAR.
+If only one register is available, the low-order bits are in the GAR and the high-order bits are on the stack, and passed on the stack if no GAR is available.
+
+.. Only floating-point members.
+
+... i.e. The structure has one long double member or one double member and two adjacent float members or 3-4 float members.
```

Comment:
```suggestion
... i.e. The structure has one long double member or one double member and two adjacent floating-point members or 3-4 floating-point members.
```

Reply:
Actually, I want to express data type `float` here. As I kown, floatting-point include data type `float` and `double`... Is my understanding right?

Reply:
> Actually, I want to express data type `float` here. As I kown, floatting-point refers to data type `float` and `double`... Is my understanding right?

I think you could make use of inline code formatting to distinguish between `data types` and general mentions.

Reply:
Okay, I just thought of it...